src/movepick.cpp

```diff
@@ -56,6 +55,8 @@ enum Stages {
     QCAPTURE
 };
 
+constexpr int BADCAPTURE = 8888888, ILLEGALMOVE = 9999999;
```

Comment:
i don't like using fixed constants for certain move types.. what else can we do?

Reply:
Hm, good question.
Dont' have any valuable idea I have for the moment....

Reply:
For illegal moves we could set the move itself to Move::none()
For bad captures we could swith them with the good captures in order to preserve all generated moves...
will try it when it find some spare time.